lasagne/layers/shape.py

```diff
@@ -32,35 +39,42 @@ class ReshapeLayer(Layer):
     A layer reshaping its input tensor to another tensor of the same total
     number of elements.
 
-    :parameters:
-        - incoming : a :class:`Layer` instance or a tuple
-            the layer feeding into this layer, or the expected input shape
-
-        - shape : tuple
-            The target shape specification. Any of its elements can be `[i]`,
-            a single-element list of int, denoting to use the size of the ith
-            input dimension. At most one element can be `-1`, denoting to
-            infer the size for this dimension to match the total number of
-            elements of the input tensor. Any remaining elements must be
-            positive integers directly giving the size of the corresponding
-            dimension.
-
-    :usage:
-        >>> from lasagne.layers import InputLayer, ReshapeLayer
-        >>> l_in = InputLayer((None, 100, 20))
-        >>> l1 = ReshapeLayer(l_in, ([0], [1], 2, 10))
-        >>> l1.get_output_shape()
-        (None, 100, 2, 10)
-        >>> l2 = ReshapeLayer(l_in, ([0], 1, 2, 5, -1))
-        >>> l2.get_output_shape()
-        (None, 1, 2, 5, 200)
-
-    :note:
-        The tensor elements will be fetched and placed in C-like order. That
-        is, reshaping `[1,2,3,4,5,6]` to shape `(2,3)` will result in a matrix
-        `[[1,2,3],[4,5,6]]`, not in `[[1,3,5],[2,4,6]]` (Fortran-like order),
-        regardless of the memory layout of the input tensor. For C-contiguous
-        input, reshaping is cheap, for others it may require copying the data.
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    shape : tuple
+        The target shape specification. In the specification integer values
+        specify output shapes and `[i]` can be used to refer to the ith
+        dimension of the input. `[i]` must be a single element list with an
+        integer value.  At most one element can be `-1`, denoting to
+        infer the size for this dimension to match the total number of
+        elements of the input tensor. Any remaining elements must be
+        positive integers directly giving the size of the corresponding
+        dimension.
+
+    Usage
+    ----------
+    >>> from lasagne.layers import InputLayer, ReshapeLayer
+    >>> l_in = InputLayer((32, 100, 20))
+    >>> l1 = ReshapeLayer(l_in, ((32, 50, 40)))
+    >>> l1.get_output_shape()
```

Comment:
This usage example is being run as a doctest and it seems to be failing (according to Travis). I think you need to add the expected output for the `get_output_shape()` calls.